### PR TITLE
Adds support for custom user model deserialization from UserManager

### PR DESCRIPTION
**Pull request by alexfdezsauco**
Adds support for custom user model deserialization from UserManager

### CODE REVIEW

File: src/Blorc.OpenIdConnect/Services/UserManager.cs [hunk 1]
```diff
@@ -41,6 +44,40 @@ public UserManager(IJSRuntime jsRuntime, NavigationManager navigationManager, IC
...
+
+        public async Task<TUser> GetUserAsync<TUser>(bool reload = true, JsonSerializerOptions options = null)
+        {
+            var userType = typeof(TUser);
+            if (!reload && _usersCache.TryGetValue(userType,  out var value) && value is TUser user)
+            {
+                return user;
+            }
+
+            if (reload)
```
Comment (GeertvanHorrik) [inline src/Blorc.OpenIdConnect/Services/UserManager.cs:66]: Why not move reload to the top? Then we don't have to check it in the if statement above.

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]
See comment. Please also add unit tests.

**Comment by alexfdezsauco** [comment]
> See comment. Please also add unit tests.

Added some test for custom user model deserialization

File: src/Blorc.OpenIdConnect.DemoApp/Models/Profile.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+﻿namespace Blorc.OpenIdConnect.DemoApp.Models
```
Comment (GeertvanHorrik) [inline src/Blorc.OpenIdConnect.DemoApp/Models/Profile.cs:1]: Please remove .Models namespace

File: src/Blorc.OpenIdConnect.DemoApp/Models/User.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿namespace Blorc.OpenIdConnect.DemoApp.Models
```
Comment (GeertvanHorrik) [inline src/Blorc.OpenIdConnect.DemoApp/Models/User.cs:1]: Please remove .Models namespace

File: src/Blorc.OpenIdConnect/Services/UserManager.cs [hunk 4]
```diff
@@ -41,6 +45,41 @@ public UserManager(IJSRuntime jsRuntime, NavigationManager navigationManager, IC
...
+            return await GetUserAsync<TUser>(options: options);
+        }
+
+        public async Task<TUser> GetUserAsync<TUser>(bool reload = true, JsonSerializerOptions options = null)
+        {
+            var userType = typeof(TUser);
+            if (reload)
+            {
+                _usersCache.Remove(userType, out _);
+                var userJsonElement = await GetUserJsonElementAsync();
```
Comment (GeertvanHorrik) [inline src/Blorc.OpenIdConnect/Services/UserManager.cs:65]: this part should be moved outside of the if (reload) now

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]

**Comment by alexfdezsauco** [comment]
This PR is completed

File: src/Blorc.OpenIdConnect/Services/UserManager.cs [hunk 5]
```diff
@@ -41,6 +42,42 @@ public UserManager(IJSRuntime jsRuntime, NavigationManager navigationManager, IC
...
+
+        public async Task<TUser> GetUserAsync<TUser>(bool reload = true, JsonSerializerOptions options = null)
+        {
+            var userType = typeof(TUser);
+            
+            if (reload)
+            {
+                _usersCache.Remove(userType, out _);
+            }
+            else if (_usersCache.TryGetValue(userType,  out var value) && value is TUser cacheUser)
```
Comment (GeertvanHorrik) [inline src/Blorc.OpenIdConnect/Services/UserManager.cs:64]: I think it's still too complex for what it's doing.

```
if (reload)
{
    _usersCache.Remove(userType, out _);
}

if (!_userCache.TryGetvalue(..., out var user))
{
    user = parse stuff

    add to cache
}

return user;
```

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]

**Review by GeertvanHorrik — APPROVED** [review]

**Pull request merged**